### PR TITLE
JS-DOM

### PR DESCRIPTION
**Pull request by mazvira**
### JS-DOM

[Demo](https://mazvira.github.io/JS-DOM) |
[Code base](https://github.com/mazvira/JS-DOM)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/mazvira/js-dom/index.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            <li class="active">Samsung</li>
+            <li>Iphone 6</li>
+            <li>Iphone XR</li>
+            <li>Huawei</li>
+        </ul>
+    </aside>
+    <div class="main"></div>
+</section>
+</body>
+</html>
```
Comment (AMashoshyna) [inline submissions/mazvira/js-dom/index.html]: Please add newlines at the end of every file.

File: submissions/mazvira/js-dom/js/scripts.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+document.addEventListener("DOMContentLoaded", function(event) {
+    var phones = [
```
Comment (AMashoshyna) [inline submissions/mazvira/js-dom/js/scripts.js]: Please use ES6 features (`let`/`const` behave more predictably than `var`s).

File: submissions/mazvira/js-dom/js/scripts.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            name: 'Samsung',
+            info: 'The Samsung Galaxy Note 8 is an Android phablet smart phone designed, developed and marketed by Samsung Electronics. Unveiled on 23 August 2017, it is the successor to the discontinued Samsung Galaxy Note 7. It became available on 15 September 2017. The Note 8 improves on the core device specifications and hallmark S Pen features of earlier devices. While retaining the same overall look and approximate size of the Galaxy S8+, it features an upgraded processor and, for the first time in Samsung\'s smartphone history, a dual-camera system on the rear of the device; one functions as a wide-angle lens and the other as a telephoto lens, with both featuring 12 MP resolution and optical image stabilization. The S Pen has increased pressure sensitivity levels and its software has been upgraded to offer improved note taking capabilities on the always-on display, as well as animated GIF and improved translation features.',
+            img: '<img src="img/samsung1.png">'
```
Comment (AMashoshyna) [inline submissions/mazvira/js-dom/js/scripts.js]: Mixing data (texts and image links in this case) with representation (html tags) is inflexible. Often you receive data from server and if you decide to change your layout (for example add CSS class to `img` element, you will need to change both frontend code and backend code. Please separate.

File: submissions/mazvira/js-dom/js/scripts.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    ];
+var li = document.querySelectorAll('li');
+var main = document.querySelector('.main');
+var ul = document.querySelector('ul');
+
+main.innerHTML = phones[0].info + phones[0].img;
+
+ul.addEventListener('click', function(e) {
+var phone = phones.find(obj => obj.name === e.target.textContent);
+main.innerHTML = phone.info + phone.img;
```
Comment (AMashoshyna) [inline submissions/mazvira/js-dom/js/scripts.js]: You can use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for string concatenation.

File: submissions/mazvira/js-dom/js/scripts.js (hunk 3 above)
Comment (A-Ostrovnyy) [inline submissions/mazvira/js-dom/js/scripts.js]: Add and fill `alt` attribute for images.

**Comment by mazvira** [comment]
Please review again after changes)

**Comment by A-Ostrovnyy** [comment]
Approved from my side. Also don't forget to mark resolved conversations.
@AMashoshyna your turn.)

File: submissions/mazvira/js-dom/js/scripts.js [hunk 5]
```diff
@@ -1,38 +1,52 @@
...
-            img:'<img src="img/huawei4.png">'
+            info: 'The P20 is constructed with a metal chassis and glass backing, has a fingerprint reader on the front that can also be used for gesture navigation, and is available in various color finishes. Both the P20 and P20 Pro feature "FullView" displays with a "notch" tab at the top-centre, with the P20 having a 5.8-inch LCD panel and the P20 Pro a 6.1-inch OLED panel. Both use a 1080p resolution with an 18.7:9 aspect ratio. Both models include a Kirin 970 system-on-chip and 128 GB of non-expandable internal storage, with 4 GB of RAM for the base model and 6 for the Pro. The models feature 3320 and 4000 mAh batteries respectively; the P20 line does not support wireless charging. Both models have a USB-C port, but the P20 Pro does not include a headphone jack.',        
         },
 
     ];
-var li = document.querySelectorAll('li');
-var main = document.querySelector('.main');
-var ul = document.querySelector('ul');
+	
+    const phonesImg = [
```
Comment (AMashoshyna) [inline submissions/mazvira/js-dom/js/scripts.js]: Not quite what I mean. You can keep phone data and sources to images together, but remove html tags from `img` property. `img` element should be built with JavaScript.

File: submissions/mazvira/js-dom/js/scripts.js [hunk 6]
```diff
@@ -1,38 +1,52 @@
...
+const main = document.querySelector('.main');
+const ul = document.querySelector('ul');
 
-main.innerHTML = phones[0].info + phones[0].img;
+main.innerHTML = [phones[0].info, phonesImg[0].img].join("");
 
 ul.addEventListener('click', function(e) {
-var phone = phones.find(obj => obj.name === e.target.textContent);
-main.innerHTML = phone.info + phone.img;
+const phone = phones.find(obj => obj.name === e.target.textContent);
```
Comment (AMashoshyna) [inline submissions/mazvira/js-dom/js/scripts.js]: please make indentations consistent throughout the code. You can use your code editor's built-in code formatting tool.

**Comment by mazvira** [comment]
Updated JS-DOM

**Pull request merged**